pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -1863,8 +1863,17 @@ public boolean hasMessageAvailable() throws PulsarClientException {
             }
 
             if (hasMoreMessages(lastMessageIdInBroker, startMessageId, resetIncludeHead)) {
-                booleanFuture.complete(true);
-                return booleanFuture;
+                //this situation will occur when :
+                // 1.We haven't read yet 2.The connection was reset multiple times
+                // 3.Broker has pushed messages to ReceiverQueue, but messages were cleaned due to connection reset
+                seekAsync(startMessageId).whenComplete((ignore, e) -> {
+                    if (e != null) {
+                        booleanFuture.complete(false);
```

Comment:
Couldn't we have the case here where the seekAsync fails and we return `false` for `hasMessageAvailable()`, when in reality there may be some message?